benchmark/querystring/querystring-parse.js

```diff
@@ -12,42 +12,48 @@ var inputs = {
   multivalue: 'foo=bar&foo=baz&foo=quux&quuy=quuz',
   multivaluemany: 'foo=bar&foo=baz&foo=quux&quuy=quuz&foo=abc&foo=def&' +
                   'foo=ghi&foo=jkl&foo=mno&foo=pqr&foo=stu&foo=vwxyz',
-  manypairs: 'a&b&c&d&e&f&g&h&i&j&k&l&m&n&o&p&q&r&s&t&u&v&w&x&y&z'
+  manypairs: 'a&b&c&d&e&f&g&h&i&j&k&l&m&n&o&p&q&r&s&t&u&v&w&x&y&z',
+  manyblankpairs: '&&&&&&&&&&&&&&&&&&&&&&&&',
+  altspaces: 'foo+bar=baz+quux&xyzzy+thud=quuy+quuz&abc=def+ghi'
 };
 
 var bench = common.createBenchmark(main, {
   type: Object.keys(inputs),
   n: [1e6],
 });
 
+// A deopt followed by a reopt of main() can happen right when the timed loop
+// starts, which seems to have a noticeable effect on the benchmark results.
+// So we explicitly disable optimization of main() to avoid this potential
+// issue.
+v8.setFlagsFromString('--allow_natives_syntax');
+eval('%NeverOptimizeFunction(main)');
+
 function main(conf) {
   var type = conf.type;
   var n = conf.n | 0;
   var input = inputs[type];
+  var i;
 
-  // Force-optimize querystring.parse() so that the benchmark doesn't get
-  // disrupted by the optimizer kicking in halfway through.
-  v8.setFlagsFromString('--allow_natives_syntax');
-  if (type !== 'multicharsep') {
-    querystring.parse(input);
-    eval('%OptimizeFunctionOnNextCall(querystring.parse)');
-    querystring.parse(input);
-  } else {
-    querystring.parse(input, '&&&&&&&&&&');
-    eval('%OptimizeFunctionOnNextCall(querystring.parse)');
-    querystring.parse(input, '&&&&&&&&&&');
-  }
+  // Note: we do *not* use OptimizeFunctionOnNextCall() here because currently
+  // it causes a deopt followed by a reopt, which could make its way into the
+  // timed loop. Instead, just execute the function a "sufficient" number of
+  // times before the timed loop to ensure the function is optimized just once.
+  if (type === 'multicharsep') {
+    for (i = 0; i < n; i += 1)
```

Comment:
Originally I used `n / 2`, but I seemed to get better (less variable) results with `n`, especially considering the default value of `n`.